### PR TITLE
Longest increasing subsequence greedy

### PR DESCRIPTION
**Pull request by anirudnits**
PR in reference to issue #460.

### CODE REVIEW

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
We already have a non-greedy implementation [here](https://github.com/TheAlgorithms/Go/blob/master/dynamic/longestincreasingsubsequence.go). Since the algorithm is similar, move it to the `github.com/TheAlgorithms/Go/dynamic/` package

**Comment by anirudnits** [comment]
Hello @raklaptudirm, I didn't want this greedy implementation under the dynamic-programming directory as it causes confusion and also is not in-line with how the repository is organized. Should I create a new directory for greedy approaches and put this in there?

**Comment by raklaptudirm** [comment]
There are no sub-packages under `dynamic`, so I don't think we should introduce one. @tjgurwara99 @siriak What are your opinions?

File: other/longestincreasingsubsequencegreedy/longestincreasingsubsequencegreedy.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package longestincreasingsubsequencegreedy
```
Comment (tjgurwara99) [inline other/longestincreasingsubsequencegreedy/longestincreasingsubsequencegreedy.go:1]: This doesn't need a separate package. There is a `dynamic` package in the repository which already has a longest increasing subsequence algorithm. Just move this algorithm there along with the old implementation.

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]
Once moved, you can also use the tests there.

File: other/longestincreasingsubsequencegreedy/longestincreasingsubsequencegreedy.go (hunk 1 above)
Comment (anirudnits) [reply other/longestincreasingsubsequencegreedy/longestincreasingsubsequencegreedy.go]: Done.

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: dynamic/longestincreasingsubsequence_test.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/TheAlgorithms/Go/dynamic"
 )
 
-func TestLongestIncreasingSubsequence(t *testing.T) {
+func LongestIncreasingSubsequenceTest(t *testing.T, algorithm func(nums []int) int) {
```
Comment (tjgurwara99) [inline dynamic/longestincreasingsubsequence_test.go:10]: This shouldn't be exported

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

**Review by tjgurwara99 — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**